docs/Using-Fleet/MDM-macOS-setup.md

```diff
@@ -0,0 +1,120 @@
+# macOS setup
+
+In Fleet, you can customize the first-time macOS setup experience for your end users:
+
+You can customize the macOS Setup Assistant and choose to show or hide specific panes.
+* Require end users to authenticate with your identity provider (IdP) and agree to an end user license agreement (EULA) before they can use their new Mac
+
+* Customize the macOS Setup Assistant by choosing to show or hide specific panes
+
+* Bootstrap new Macs with your configuration management tool of choice (ex. Munki, Chef, or Puppet)
+
+## End user authentication
+
+> This feature is currently in development.
+
+## Bootstrap package
+
+_Available in Fleet Premium_
+
+Fleet supports installing a single package on macOS hosts that automatically enroll to Fleet.
+
+> In addition to installing the bootstrap package, Fleet automatically installs the fleetd agent on hosts that automatically enroll. This agent is responsible for reporting host vitals to Fleet and presenting Fleet Desktop to the end user.
+
+To add a bootstrap package to Fleet, we will do the following steps:
+
+1. Download or generate a package
+2. Sign the package
+3. Upload the package to Fleet
+
+### Step 1: Download or generate a package
+
+If you use Munki, Chef, Puppet, or another configuration management tool, download the client (agent) for your tool. You can find the client on each tool's GitHub or website. For example, you can download Munki, the Munki client on their [releases page on GitHub](https://github.com/munki/munki/releases). 
+
+Make sure the file you download is a `.pkg` file.
+
+If you plan to run a custom script during macOS setup, you'll need to generate a package. The [munkipkg tool](https://github.com/munki/munki-pkg) is a popular tool for generating packages.
+
+Make sure the package you generate is a `.pkg` file.
+
+### Step 2: Sign the package
+
+1. Obtain an appropriate TLS/SSL certificate with signing capability. You can do this by:
+   - Using an [Apple developer account](https://developer.apple.com/account).
+     - [Link your developer account to
+       Xcode](https://help.apple.com/xcode/mac/current/#/dev154b28f09) or [using your developer account](https://developer.apple.com/help/account/create-certificates/create-developer-id-certificates).
+     - Ensure you choose "Developer ID Installer" as the certificate type.
+     - Verify the certificate is saved to your macOS Keychain.
+
+   - Acquiring a certificate from third parties that meet the requirements.
+     - Follow their instructions for creating a TLS/SSL certificate with signing usage.
+     - Add the resulting `.p12` to your keychain.
+2. Sign your `pkg` with a valid Developer ID certificate:\
+\
+```productsign --sign "Developer ID Installer: Your Developer Name (SerialNumber)" /path/to/your.pkg /path/to/your-signed.pkg``` \
+
+3. Upload the package to Apple's notary service for validation: \
```

Comment:
This might only apply to system extensions (formerly kernel extensions), but I was under the impression that all apps and pkgs need to be notarized but not stapled. [https://developer.apple.com/developer-id/](https://developer.apple.com/developer-id/)  I'll see if I can find more information to confirm.

Reply:
Thanks! 

Yeah, I did a quick "installenterpriseapplication notarize" search in MacAdmins slack and found a couple messages that say you don't have to notarize. For example: https://macadmins.slack.com/archives/C19MR7EM9/p1633637758046800

I think we should test this. @roperzh, for when you're back, do you know if the bootstrap package needs to be notarized?

Reply:
@noahtalerman It doesn't need to be notarized. Two things are key though, otherwise the installation fails:

1. It needs to be signed with a valid developer ID certificate
2. It needs to be a "distribution" style package

`2` depends on how you're building your package. Tools like Munki and [macadmins/installapplications](https://github.com/macadmins/installapplications) (`installapplications` uses Munki) allow you to set up that easily.

I can give you more details about how to do it if it's helpful.

Reply:
@roperzh welcome back! 

Do the docs cover the two key things you outlined above? https://fleetdm.com/docs/using-fleet/mdm-macos-setup#bootstrap-package

Reply:
thank you!

> Do the docs cover the two key things you outlined above? https://fleetdm.com/docs/using-fleet/mdm-macos-setup#bootstrap-package

I think it doesn't **explicitly** cover `2`. I haven't been able to find official documentation, but this is a good summary: https://scriptingosx.com/2017/09/on-distribution-packages/

In short, the package needs to be built in a "special" way.